### ISSUE TITLE
please provide default for includeSynthToNameHashing

### ISSUE THREAD

**Issue by fommil** [sbt/sbt#2584]
we have something like this in `sbt-ensime` for `compileOnly`

``` scala
  private object noopCallback extends xsbti.AnalysisCallback {
    override val nameHashing: Boolean = true
    def beginSource(source: File): Unit = {}
    def generatedClass(source: File, module: File, name: String): Unit = {}
    def api(sourceFile: File, source: xsbti.api.SourceAPI): Unit = {}
    def sourceDependency(dependsOn: File, source: File, publicInherited: Boolean): Unit = {}
    def binaryDependency(binary: File, name: String, source: File, publicInherited: Boolean): Unit = {}
    def endSource(sourcePath: File): Unit = {}
    def problem(what: String, pos: xsbti.Position, msg: String, severity: xsbti.Severity, reported: Boolean): Unit = {}
    def usedName(sourceFile: File, names: String): Unit = {}
    override def binaryDependency(file: File, s: String, file1: File, dependencyContext: xsbti.DependencyContext): Unit = {}
    override def sourceDependency(file: File, file1: File, dependencyContext: xsbti.DependencyContext): Unit = {}
  }
```

but it fails at runtime without this being added:

```
    val includeSynthToNameHashing: Boolean = true
```

because the API has changed. Can you please add a default so that existing code extending this interface is still valid?

**Comment by jvican** [comment]
I'm a little bit confused here. Is this for sbt 1.0 or 0.13.x? Why doesn't the compiler fail if you don't implement `includeSynthToNameHashing`? That method is unimplemented in the Java interface defined in `xsbti.AnalysisCallback` in sbt 1.0.

**Comment by fommil** [comment]
It was for 0.13

**Comment by jvican** [comment]
So since we're gonna focus on sbt 1.0, can we close this issue? As far as I know, this is fixed in sbt 1.0 since you would not be able to create a callback without defining `includeSynthToNameHashing`.
